### PR TITLE
[API] Hide Mail-Address if the user wishes to

### PR DESCRIPTION
**Pull request by NobodysNightmare**
## OpenProject work package

https://community.openproject.org/work_packages/19340

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://nobodysnightmare-feature-hide-mail-address.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://nobodysnightmare-feature-hide-mail-address.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

**Comment by ulferts** [comment]
@NobodysNightmare the renaming is not reflected in the front end tests: tests/integration/mocks/work-packages/... I don't think that the mail address is used anywhere (and you definitely have tested this with the front end, right) but this should be adapted.

**Comment by NobodysNightmare** [comment]
@ulferts There are no parts where I know the mail address being used, thus I could not try out if anything breaks.

However, you are right: I fixed the mocks and ideally a test would fail if the `mail` field was already used.

File: doc/apiv3-documentation.apib [hunk 1]
```diff
@@ -2152,18 +2152,18 @@ This endpoint lists the types that are *available* in a given project.
...
+| lastName    | User's last name                                          | String   |             | READ                 |                           |
+| name        | User's full name, formatting depends on instance settings | String   |             | READ                 |                           |
+| email       | User's E-Mail address                                     | String   |             | READ                 | E-Mail address not hidden |
+| avatar      | URL to user's avatar                                      | String   |             | READ                 |                           |
+| status      | The current activation status of the user (see below)     | String   |             | READ                 |                           |
+| createdAt   | Time of creation                                          | DateTime |             | READ                 |                           |
+| updatedAt   | Time of the most recent change to the user                | DateTime |             | READ                 |                           |
```
Comment (Azure7111) [inline doc/apiv3-documentation.apib:2166]: You might want to lead by example below ;)

**Pull request merged**